### PR TITLE
Fix first load of CDAP UI in k8s environment

### PR DESCRIPTION
**Pull request by ajainarayanan**
cherry-pick of https://github.com/cdapio/cdap/pull/13046

### CODE REVIEW

File: cdap-ui/server.js [hunk 1]
```diff
@@ -157,7 +158,8 @@ getCDAPConfig()
     sockServer.on('connection', function(c) {
       log.debug('[SOCKET OPEN] Connection to client "' + c.id + '" opened');
       // @ts-ignore
-      var a = new Aggregator(c);
+      var a = new Aggregator(c, { ...cdapConfig, ...securityConfig });
+      wsConnections.push(c);
```
Comment (elfenheart) [inline cdap-ui/server.js:162]: we should remove this connection on close

File: cdap-ui/server.js [hunk 2]
```diff
@@ -171,6 +173,21 @@ getCDAPConfig()
         log.info('Denying socket connection and closing the channel');
         socket.end();
         socket.destroy();
+        return;
       }
     });
+    function gracefulShutdown() {
+      log.info('Caught SIGTERM. Closing http & ws server');
+      server.close();
+      if(Array.isArray(wsConnections) && wsConnections.length) {
```
Comment (elfenheart) [inline cdap-ui/server.js:183]: nit: space after `if`

File: cdap-ui/server.js (hunk 1 above)
Comment (ajainarayanan) [reply cdap-ui/server.js:162]: @elfenheart I am doing it on SIGTERM handler (L184). Unless I am missing something

Comment (chtyim) [reply cdap-ui/server.js:162]: So we are keeping all connections in memory and only clear on SIGTERM? So if the server runs for a long time, will it OOM?

**Comment by chtyim** [comment]
BTW, the PR should be opened against develop, not release/6.3. Only open against release branch if it is cherry pick.

File: cdap-ui/server.js (hunk 1 above)
Comment (elfenheart) [reply cdap-ui/server.js:162]: SIGTERM is when the server is terminated, right? But when individual connection get closed, we need to remove them from the list.

**Review by elfenheart — APPROVED** [review]

**Pull request merged**